src/azure-cli/azure/cli/command_modules/network/tunnel.py

```diff
@@ -0,0 +1,206 @@
+        is_port_open = False
+        with closing(socket.socket(socket.AF_INET, socket.SOCK_STREAM)) as sock:
+            if sock.connect_ex(('', self.local_port)) == 0:
+                logger.info('Port %s is NOT open', self.local_port)
+            else:
+                logger.info('Port %s is open', self.local_port)
+                is_port_open = True
+            return is_port_open
+
+    def _get_auth_token(self):
+        profile = Profile(cli_ctx=self.cli_ctx)
+        user = profile.get_current_account_user()
+        # Generate an Azure token with the VSTS resource app id
+        auth_token = profile.get_access_token_for_resource(user, None, None)
+        content = {
+            'resourceId': self.remote_host,
+            'protocol': 'tcptunnel',
+            'workloadHostPort': self.remote_port,
+            'aztoken': auth_token,
+            'token': self.last_token,
+        }
+        if self.node_id:
+            custom_header = {'X-Node-Id': self.node_id}
+        else:
+            custom_header = {}
+
+        web_address = 'https://{}/api/tokens'.format(self.bastion.dns_name)
+        response = requests.post(web_address, data=content, headers=custom_header,
+                                 verify=(not should_disable_connection_verify()))
+        response_json = None
+
+        if response.content is not None:
+            response_json = json.loads(response.content.decode("utf-8"))
+
+        if response.status_code not in [200]:
+            if response_json is not None and response_json["message"] is not None:
+                exp = CloudError(response, error=response_json["message"])
+            else:
+                exp = CloudError(response)
+            raise exp
+
+        self.last_token = response_json["authToken"]
+        self.node_id = response_json["nodeId"]
+        return self.last_token
+
+    def _listen(self):
+        self.sock.setblocking(True)
+        self.sock.listen(100)
+        index = 0
+        while True:
+            self.client, _address = self.sock.accept()
+
+            auth_token = self._get_auth_token()
+            host = 'wss://{}/webtunnel/{}?X-Node-Id={}'.format(self.bastion.dns_name, auth_token, self.node_id)
+            self.ws = create_connection(host,
+                                        sockopt=((socket.IPPROTO_TCP, socket.TCP_NODELAY, 1),),
+                                        sslopt={'cert_reqs': ssl.CERT_NONE},
+                                        enable_multithread=True)
+            logger.info('Websocket, connected status: %s', self.ws.connected)
+            index = index + 1
+            logger.info('Got debugger connection... index: %s', index)
+            debugger_thread = Thread(target=self._listen_to_client, args=(self.client, self.ws, index))
+            web_socket_thread = Thread(target=self._listen_to_web_socket, args=(self.client, self.ws, index))
+            debugger_thread.start()
+            web_socket_thread.start()
```

Comment:
App Service have update the implementation of tunnel.py please update it.

https://github.com/Azure/azure-cli/blob/213997916f90666e0e70050b24ca94c17fc4ab55/src/azure-cli/azure/cli/command_modules/appservice/tunnel.py#L139-L153